x-pack/plugins/security_solution/server/lib/entity_analytics/risk_engine/risk_engine_data_client.ts

```diff
@@ -71,12 +74,18 @@ export class RiskEngineDataClient {
   private writerCache: Map<string, Writer> = new Map();
   constructor(private readonly options: RiskEngineDataClientOpts) {}
 
-  public async init({ namespace, taskManager }: InitOpts) {
+  public async init({
+    namespace,
+    taskManager,
+    assetCriticalityDataClient,
```

Comment:
It seems like `RiskEngineDataClient` was previously responsible for the risk _scoring_ resources. While asset criticality is arguably also a risk scoring resource, it's also got a dedicated client, now, but that client is consumed by `RiskEngineDataClient`.

I think the ideal architecture for me here would be a general `EntityAnalyticsDataClient` that coordinates both this `RiskScoringDataClient` and an `AssetCriticalityDataClient`, but handles them independently.

Can you explain/defend the architecture as it stands in this PR? I already had concerns about `RiskEngineDataClient` doing too much, and now we're adding more here.

Reply:
Hey, I assume that `RiskEngineDataClient` is already doing very similar things that you mention for `EntityAnalyticsDataClient`

Let's observe what `RiskEngineDataClient` doing right now:

- manage initialisation of risk engine:
  - disable legacy risk engine, by removing all dashboards and transforms
  - create configuration (Saved object)
  - creating mappings, datastreams, transforms for risk scoring
  - enable the risk engine and start the task
  - **(new)** init Aset Criticality by calling AssetCriticalityDataClient 

- enable/disable risk engine

- provide the status of the risk engine

- provide writer for calculate and persist scores

- provide risk input index


I agree that `RiskEngineDataClient` does a lot of things, and from my perspective, its ideal role is:
- manage common resources (Saved object configuration)
- install the risk engine
- provide status and enable/disable methods
- **(potentially)** health check/ health status


I think that potential future improvements are to extract creating risk score mappings, datastream, transforms, and writer - to something like `RiskScoringDataClient`

I also assume that `RiskEngineDataClient` still will accept `RiskScoringDataClient` and `AssetCriticalityDataClient` as a dependency for risk engine install. So, this is why I propose to keep it as it is for this PR and have more discussion in the future about splitting `RiskEngineDataClient`.

Reply:
I agree with the facts as you have presented them, but I don't think that we should continue to defer these discussions and accumulate debt like this. It's never going to be a convenient time to clean this up, but I think we need it and I'm asking about it now.

My broad goals are:

1. Separation of responsibilities: risk scoring assets are separate from asset criticality assets, and details of these are hidden from the parent "coordinator" (e.g. `RiskEngineDataClient`)
2. Separation of state: the parent coordinator holds any state necessary, and makes the "decisions:" e.g. a `DataClient` is just a collection of stateless methods (`getLegacyStatus`, `createIndexTemplate`), and the coordinator leverages those (e.g. calls `deleteLegacyTransforms` based on `getLegacyStatus`).